.github/workflows/tests.yml

```diff
@@ -37,6 +37,15 @@ jobs:
       with:
         python-version: ${{ matrix.python-version }}
 
+    - name: Cache dependency
+      id: cache-dependencies
+      uses: actions/cache@v4
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-pip-${{ hashFiles('**/pyproject.toml') }}
+        restore-keys: |
+          ${{ runner.os }}-pip-
```

Comment:
I think simply using this is sufficient (and simpler to understand). But this is nitpick
```suggestion
        key: ${{ runner.os }}-pip
```

Reply:
True, this will only make a difference in few corner cases. I'm fine with removing it.